src/backend/app/Models/Token.php

```diff
@@ -17,4 +56,11 @@ public function transaction(): BelongsTo {
     public function paymentHistories(): MorphOne {
         return $this->morphOne(PaymentHistory::class, 'paid_for');
     }
+
+    public function soldEnergyPerPeriod($startDate, $endDate): Builder {
+        return $this::query()
+            ->select(DB::raw(' SUM(load) as sold,YEARWEEK(created_at,3) as period'))
+            ->whereBetween('created_at', [$startDate, $endDate])
+            ->groupBy(DB::raw('YEARWEEK(created_at,3)'));
+    }
```

Comment:
This would only make sense for `kWH` Tokens, so we should add a filter here. I.e. `WHERE token_type = ...`.

However, are we even using this function at all?

Reply:
> This would only make sense for kWH Tokens, so we should add a filter here. I.e. WHERE token_type = ....

Good catch on this, will fix. 

> However, are we even using this function at all?

I will have to check, it existed for MeterToken so I just ported it.